server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -300,6 +303,24 @@ public int compare(Interval o1, Interval o2)
     return metrics;
   }
 
+  @GET
+  @Path("/{dataSourceName}/candidates/intervals/{intervals}/numCandidates/{numCandidates}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Iterable<LocatedSegmentDescriptor> getQueryTargets(
+      @PathParam("dataSourceName") String datasource,
+      @PathParam("intervals") String intervals,
+      @PathParam("numCandidates") String numCandidates,
+      @Context final HttpServletRequest req
+  ) throws IOException
+  {
+    List<Interval> intervalList = Lists.newArrayList();
+    for (String interval : intervals.split(",")) {
+      intervalList.add(Interval.parse(interval.trim()));
+    }
```

Comment:
When Intervals are specified as PathParam do we need to do any handling for '/' ? 
e.g similar thing is done in DataSourcesResource.deleteDataSourceSpecificInterval